src/reanimated2/Hooks.js

```diff
@@ -311,6 +310,13 @@ export function useAnimatedStyle(updater, dependencies) {
     };
   }, dependencies);
 
+  useEffect(() => {
+    return () => {
+      initRef.current = null;
```

Comment:
This needs some explanation. I don't see why it'd actually make any difference. When hook is unmounted the `initRef` is gone, so it will not retain reference via `current` field either.

Even if we need this behavior it'd be a lot better to have it designed in a more resilient way and have custom hook that both creates a ref and also cleans it up instead of repeating the same code over and over in five places.